hasher-matcher-actioner/hmalib/matchers/filters.py

```diff
@@ -0,0 +1,289 @@
+        match_distance: int,
+    ) -> bool:
+        if match.get_source() != THREAT_EXCHANGE_SOURCE_SHORT_CODE:
+            # Do not have an opinion on matches outside threatexchange
+            return True
+
+        te_match = t.cast(ThreatExchangeIndicatorIndexMetadata, match)
+        return get_privacy_group_matcher_active(te_match.privacy_group)
+
+
+class ThreatExchangePdqMatchDistanceFilter(BaseMatchFilter):
+    """
+    Filter out matches where the match distance is > the threshold set for that
+    privacy group.
+    """
+
+    def filter_one_metadata_obj(
+        self,
+        match: BaseIndexMetadata,
+        index_signal_type: t.Type[SignalType],
+        match_distance: int,
+    ) -> bool:
+        if match.get_source() != THREAT_EXCHANGE_SOURCE_SHORT_CODE:
+            # Do not have an opinion on matches outside threatexchange
+            return True
+
+        if index_signal_type != PdqSignal:
+            # Do not have an opinion on matches that are not pdq
+            return True
+
+        pg_matcher_threshold = get_privacy_group_matcher_pdq_threshold(
+            str(t.cast(ThreatExchangeIndicatorIndexMetadata, match).privacy_group)
+        )
+        return match_distance <= pg_matcher_threshold
+
+
+class BankActiveFilter(BaseMatchFilter):
+    """
+    Filter out matches from banks where the is_active flag is False.
+    """
+
+    def __init__(self, banks_table: BanksTable):
+        self.banks_table = banks_table
+
+    def filter_one_metadata_obj(
+        self,
+        match: BaseIndexMetadata,
+        index_signal_type: t.Type[SignalType],
+        match_distance: int,
+    ) -> bool:
+        if match.get_source() != BANKS_SOURCE_SHORT_CODE:
+            # Do not have an opinion on matches that are not from banks.
+            return True
+
+        bank_match = t.cast(BankedSignalIndexMetadata, match)
+        return self.get_bank_active(bank_match.bank_member_id)
+
+    def get_bank_active(self, bank_member_id: str) -> bool:
+        """
+        Is the bank active? Use the bank_member_id to get an answer. Note
+        bank_id -> active is cached, but bank_member_id -> bank_id is not.
+        """
+        member = self.get_bank_member_internal(
+            bank_member_id, time.time() // BANK_CACHE_TIME_SECONDS
+        )
+        return self.get_bank_active_internal(
+            member.bank_id, time.time() // BANK_CACHE_TIME_SECONDS
+        )
+
+    @methodtools.lru_cache(maxsize=BANK_CACHE_SIZE)
+    def get_bank_active_internal(self, bank_id: str, cache_buster) -> bool:
+        """
+        Retrieves is_active from the banks table. Use a time.time() // CACHE_TIME
+        style value for `cache_buster` to add a sort of TTL.
+        """
+        return self.banks_table.get_bank(bank_id=bank_id).is_active
+
+    @methodtools.lru_cache(maxsize=BANK_CACHE_SIZE)
+    def get_bank_member_internal(self, bank_member_id: str, cache_buster) -> BankMember:
```

Comment:
These had to become class methods because I needed banks_table to be passed along as constructor args. 

Privacy groups on the other hand, can be inferred from the environment because that's how HMAConfig works.